### PR TITLE
[Core] Fix Prism3D15

### LINKED ISSUE #9818: [KratosCore] Bug in Prism3D15N geometry
**Description**
The prism3D15N geometry seems to be buggy. [Here (test_Prism3D15N.zip)](https://github.com/KratosMultiphysics/Kratos/files/8399186/test_Prism3D15N.zip) you can find a minimal test case. It is a structural modal analysis with one SmallDisplacementElement3D15N which uses this prism. The single included element follows the Kratos node ordering specified [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/geometries/prism_3d_15.h#L36). However, if one runs the analysis, it throws an error saying that the element base class is called. From the other output, it seems that something is very wrong with the computation of the Jacobian or the shape function (derivatives). 

Now, I checked, if there is a test for this geometry. However, it seems not. In fact, a corresponding test was once committed but then commented out. @loumalouomega from the git history, I noticed that you commented it out saying in the title of the [corresponding commit](https://github.com/KratosMultiphysics/Kratos/commit/bc7faa704785deb36125f50d04d0914824ae0dab) that "Prism3D15 [is](not working yet, the element has some problems to be fixed in the future)".

So my questions is. Did anybody experience this bug before or know a solution for it? It would be great if anyone involved in the geometry implementation could have a look into the element and check if there is something fishy.

Side note: If one changes the prism3D15N to a prism3D6N, the minimal example attached runs again without errors (relevant mdpas are provided). So it just seems to be a bug of the quadratic prism (and not the solver / model setup, etc...).  

**Scope**
- KratosCore

**To Reproduce**
Minimal example attached

**Expected behavior**
The simulation (a modal analysis) should run through and print the eigenvalue of the first mode

**Environment**
- OS: Windows
- Branch: master
- Python 3.9.7

### PR DESCRIPTION
**Pull request by RiccardoRossi**
**📝 Description**
fixes the calculation of shape functons and of their derivatives for the Prism3D15
fixed #9818 

Please mark the PR with appropriate tags: 
- Hotfix

**🆕 Changelog**
fixes the calculation of shape functons and of their derivatives for the Prism3D15
fixed #9818

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
posing here the code used to generate the shape functions. It may be useful in the future...

~~~py
import numpy as np
from sympy import *

x = Symbol("x")
y = Symbol("y")
z = Symbol("z")

def LinearTriangleLocalShapeFunctions(x,y):
    t = 1.0 - x - y;
    Nlocal = [    
        t,
        x,
        y
    ]
    return Nlocal


def TriangleLocalShapeFunctions(x,y):
    t = 1.0 - x - y;
    Nlocal = [    
        t * (2.0 * t - 1.0),
        x * (2.0 * x - 1.0),
        y * (2.0 * y - 1.0),
        4.0 * t * x,
        4.0 * x * y,
        4.0 * y * t
    ]
    return Nlocal

print("(0.5,0.5)=",TriangleLocalShapeFunctions(0.5,0.5))
print("(-1.0,1.0)=",TriangleLocalShapeFunctions(-1.0,1.0))
print("(1.0,-1.0)=",TriangleLocalShapeFunctions(1.0,-1.0))
print("(0.0,0.0)=",TriangleLocalShapeFunctions(.0,.0))
print("(1.0,0.0)=",TriangleLocalShapeFunctions(1.0,0.0))

def LineLocalShapeFunctions(z):
    zaux = z*2-1
    Nlocal = [
        (zaux - 1.0) * zaux / 2,
        (zaux + 1.0) * zaux / 2,
        1.0 - zaux * zaux
    ]
    return Nlocal

print(LineLocalShapeFunctions(0))
print(LineLocalShapeFunctions(1))
print(LineLocalShapeFunctions(0.5))
print(LineLocalShapeFunctions(z))

#  *                 3
#  *               ,/|`\
#  *             12  |  14
#  *           ,/    |    `\
#  *          4------13-----5
#  *          |      9      |
#  *          |      |      |
#  *          |      |      |
#  *          |      |      |
#  *          10     |      11
#  *          |      0      |
#  *          |    ,/ `\    |
#  *          |  ,6     `8  |
#  *          |,/         `\|
#  *          1------7------2
def PrismShapeFunctions(x,y,z):
    Ntri = TriangleLocalShapeFunctions(x,y)
    Ntri_linear = LinearTriangleLocalShapeFunctions(x,y)
    Nline = LineLocalShapeFunctions(z)
    N = [
        Ntri[0]*Nline[0], #0 #vertex nodes of bottom face 
        Ntri[1]*Nline[0], #1
        Ntri[2]*Nline[0], #2
        Ntri[0]*Nline[1], #3 #vertex nodes of top face
        Ntri[1]*Nline[1], #4
        Ntri[2]*Nline[1], #5
        Ntri[3]*Nline[0], #6 #mid nodes of bottom face
        Ntri[4]*Nline[0], #7
        Ntri[5]*Nline[0], #8
        Ntri_linear[0]*Nline[2], #9 #vertex nodes of lateral edges
        Ntri_linear[1]*Nline[2], #10
        Ntri_linear[2]*Nline[2], #11
        Ntri[3]*Nline[1], #12 #mid nodes of top face
        Ntri[4]*Nline[1], #13
        Ntri[5]*Nline[1], #14
        ]
    return N

def PrismShapeFunctionsDerivatives(x,y,z):
    N = PrismShapeFunctions(x,y,z)
    DN = zeros(15,3)
    for i in range(15):
        DN[i,0] = simplify(diff(N[i],x))
        DN[i,1] = simplify(diff(N[i],y))
        DN[i,2] = simplify(diff(N[i],z)) 
    return DN

Ntri = TriangleLocalShapeFunctions(x,y)
Nsum = 0
for n in Ntri:
    Nsum += n 
Nsum = simplify(Nsum)
print("Ntriangle sum = ",Nsum)

Nline = LineLocalShapeFunctions(z)
Nsum = 0
for n in Nline:
    Nsum += n 
Nsum = simplify(Nsum)
print("Nline sum = ",Nsum)

# x = 0.5
# y = 0.5
# z = 0.5
N = PrismShapeFunctions(x,y,z)
print(N)
#verify that the sum of N is 1
Nsum = 0
for n in N:
    Nsum += n 
#Nsum = simplify(Nsum.subs({x:0.3,y:0.4,z:0.55}))
print(simplify(Nsum))

print([simplify(n.subs({x:0,y:0.5,z:-1})) for n in N])

DN = PrismShapeFunctionsDerivatives(x,y,z)
for k in range(3):
    Nsum = 0
    for i in range(15):
        Nsum += DN[i,k]
    print(simplify(Nsum))

for i in range(15):
    print("case",i,": return ",ccode(N[i]),";")

for i in range(15):
    print("shape_function_values( pnt, ",i," ) =",ccode(N[i]),";")


for i in range(15):
    for k in range(3):
        print("DN(",i,",",k,")=",ccode(DN[i,k]),";")

~~~

File: kratos/utilities/geometry_tester.h [hunk 1]
```diff
@@ -130,7 +130,7 @@ class GeometryTesterUtility
         if(TestHexahedraInterface3D8N(model_part) == false) succesful=false;
 
         if(TestPrism3D6N(model_part) == false) succesful=false;
-//        if(TestPrism3D15N( error_msg ) == false) succesful=false;
+//         if(TestPrism3D15N(model_part) == false) succesful=false;
```
Comment (loumalouomega) [inline kratos/utilities/geometry_tester.h]:
```suggestion
         if(TestPrism3D15N(model_part) == false) succesful=false;
```
This should work now

**Review by loumalouomega — DISMISSED** [review]
Thanks

**Review by loumalouomega — DISMISSED** [review]

**Review by loumalouomega — DISMISSED** [review]

File: kratos/utilities/geometry_tester.h [hunk 2]
```diff
@@ -130,7 +130,7 @@ class GeometryTesterUtility
         if(TestHexahedraInterface3D8N(model_part) == false) succesful=false;
 
         if(TestPrism3D6N(model_part) == false) succesful=false;
-//        if(TestPrism3D15N( error_msg ) == false) succesful=false;
+         if(TestPrism3D15N(model_part) == false) succesful=false;
```
Comment (loumalouomega) [inline kratos/utilities/geometry_tester.h]:
```suggestion
        //if(TestPrism3D15N(model_part) == false) succesful=false;  TODO: Debug this
```

**Review by loumalouomega — APPROVED** [review]

**Comment by RiccardoRossi** [comment]
@loumalouomega I let you the the honor of merging this! ...after all this was your branch!! :-)

**Pull request merged**